linera-storage/src/lib.rs

```diff
@@ -424,6 +424,14 @@ where
     async fn contains_blob(&self, blob_id: BlobId) -> Result<bool, ViewError> {
         self.storage.contains_blob(blob_id).await
     }
+
+    async fn add_blob(&self, blob: Blob) -> Result<(), ViewError> {
+        if cfg!(with_testing) {
+            self.storage.write_blob(&blob).await
+        } else {
+            unimplemented!()
```

Comment:
That looks a bit dangerous. Can't we just make the whole method `#[cfg(with_testing)]`?

Reply:
That's what I first tried, but then CI fails on `cargo check-all-features` because when we try to check other crates with the `test` feature, it won't match the `test` feature from `linera-storage`, then even though we implement `add_blob` under the `test` feature, it won't find that implementation. This way at least that'll compile 😅 but I agree it seems weird to have to do that 🤔
Not sure what else to do though if we want to keep checking that on CI

Reply:
That's a bug in our `Cargo.toml`s then: Other crates with the `test` feature should always depend on `linera-storage/test`.

Reply:
Ah! I did not think of that, good point. Let me check